indico/web/forms/fields.py

```diff
@@ -540,3 +541,22 @@ def __init__(self, *args, **kwargs):
 
     def _value(self):
         return self.text_value
+
+
+class IndicoColorSwitchField(JSONField):
+    """Field allowing user to pick a color from a set of predefined values"""
+
+    widget = JinjaWidget('forms/colorswitch_widget.html')
+    default_widget_colors = {'background': 'EEE0EF', 'text': '1D041F'}
+
+    def process_formdata(self, valuelist):
+        super(IndicoColorSwitchField, self).process_formdata(valuelist)
+        self.data = ColorTuple(self.data['text'], self.data['background'])
+
+    def populate_obj(self, obj, name):
+        obj.colors = self.data
```

Comment:
Not a good idea. We shouldn't assume a certain name (even though it'll probably always be `colors`) - what's wrong with the default population code that uses the field name?